store/tikv/mock-tikv/mvcc.go

```diff
@@ -166,17 +166,22 @@ func (e *mvccEntry) Commit(startTS, commitTS uint64) error {
 
 func (e *mvccEntry) Rollback(startTS uint64) error {
 	if e.lock == nil || e.lock.startTS != startTS {
-		if commitTS, ok := e.checkTxnCommitted(startTS); ok {
-			return ErrAlreadyCommitted(commitTS)
+		c, ok := e.getTxnCommitInfo(startTS)
+		if ok && c.valueType != typeRollback {
+			return ErrAlreadyCommitted(c.commitTS)
 		}
-		return nil
+		if ok && c.valueType == typeRollback {
+			return nil
+		}
+	} else {
+		e.lock = nil
 	}
+
 	e.values = append([]mvccValue{{
 		valueType: typeRollback,
 		startTS:   startTS,
 		commitTS:  startTS,
 	}}, e.values...)
-	e.lock = nil
```

Comment:
It still needs to clean the lock if `e.lock.startTS != startTS` and `getTxnCommitInfo() returns false`.